diff --git a/helpers/z_schema.js b/helpers/z_schema.js
index 6877118..be0b43f 100644
--- a/helpers/z_schema.js
+++ b/helpers/z_schema.js
@@ -84,6 +84,12 @@ function schema(network){
     }
   });
 
+  this.z_schema.registerFormat('voteString', function (str) {
+    //Excluding capital hex characters
+    //(mainnet database could contain mixed case vote strings?)
+    return /^[-+]0[23][0-9a-f]{64}$/.test(str);
+  });
+
   this.z_schema.registerFormat('queryList', function (obj) {
     obj.limit = 100;
     return true;
diff --git a/logic/account.js b/logic/account.js
index a01f68d..d0605e9 100644
--- a/logic/account.js
+++ b/logic/account.js
@@ -17,7 +17,7 @@ function Account (scope, cb) {
 
        self = this;
        db = this.scope.db;
-       library = this.scope.library;
+       library = this.scope;
        genesisBlock = this.scope.genesisblock.block;
 
        this.table = 'mem_accounts';
diff --git a/logic/vote.js b/logic/vote.js
index e0fc8b7..77e4bd9 100644
--- a/logic/vote.js
+++ b/logic/vote.js
@@ -210,6 +210,10 @@ Vote.prototype.schema = {
        properties: {
                votes: {
                        type: 'array',
+                       items: {
+                               type: 'string',
+                               format: 'voteString',
+                       },
                        minLength: 1,
                        maxLength: constants.maximumVotes,
                        uniqueItems: true
